src/main/java/seedu/recipe/model/cooked/UniqueRecordList.java

```diff
@@ -135,17 +134,15 @@ private boolean recordsAreUnique(List<Record> records) {
     /**
      * Sets goal tally when records are set initially.
      */
-    private void setGoalsTally() {
-        HashMap<String, Integer> goalMap = new HashMap<String, Integer>();
+    public void setGoalsTally() {
```

Comment:
does the pie chart still malfunction and multiply in segments when a meal is cooked after undoing?
 Because if it still does we might have to remove undo for the cooked feature as this would break the goals feature! Not sure why it doesn't update either perhaps it has something to do with the setData function. Will check out it tomorrow and see if I can find a solution too. Thanks for the hardwork Brian!

Reply:
nope the pie chart doesn't split into further segments if a meal is cooked after undoing.

I tested these commands in this order: `cooked 5` -> `undo` -> `cooked 5`. The _before_ behavior was that the pie chart will split into more segments. The _current_ behavior is the correct, expected behavior.